src/fabric/src/fabric_view.erl

```diff
@@ -239,14 +234,15 @@ possibly_embed_doc(
                             Resp
                     end
             end;
-        _ ->
+        false ->
             Row
     end.
 
-detach_partition(#view_row{key = {p, _Partition, Key}} = Row) ->
-    Row#view_row{key = Key};
-detach_partition(#view_row{} = Row) ->
-    Row.
+detach_partition(Row) ->
+    case fabric_view_row:get_key(Row) of
+        {p, _Partition, Key} -> fabric_view_row:set_key(Row, Key);
+        _PrimitiveKey -> Row
```

Comment:
Primitive is a bit of an odd name, to avoid introducing new terminology maybe just use `_`?

Reply:
Sure.  Or we could just use `_Key`.